app/components/markdown/markdown_image.js

```diff
@@ -1,70 +1,206 @@
+import {normalizeProtocol} from 'app/utils/url';
+
+const MAX_IMAGE_HEIGHT = 150;
+
+export default class MarkdownImage extends React.Component {
     static propTypes = {
-        src: PropTypes.string.isRequired
+        children: PropTypes.node,
+        linkDestination: PropTypes.string,
+        onLongPress: PropTypes.func,
+        source: PropTypes.string.isRequired,
+        textStyle: CustomPropTypes.Style
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired
     };
 
     constructor(props) {
         super(props);
 
         this.state = {
-            width: 10000,
-            maxWidth: 10000,
-            height: 0
+            width: 0,
+            height: 0,
+            maxWidth: Math.MAX_INT,
+            failed: false
         };
+
+        this.mounted = false;
     }
 
     componentWillMount() {
-        Image.getSize(this.props.src, this.handleSizeReceived, this.handleSizeFailed);
+        this.loadImageSize(this.props.source);
+    }
+
+    componentDidMount() {
+        this.mounted = true;
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.src !== nextProps.src) {
-            Image.getSize(nextProps.src, this.handleSizeReceived, this.handleSizeFailed);
+        if (this.props.source !== nextProps.source) {
+            this.setState({
+                width: 0,
+                height: 0,
+                failed: false
+            });
+
+            this.loadImageSize(nextProps.source);
         }
     }
 
+    componentWillUnmount() {
+        this.mounted = false;
+    }
+
+    loadImageSize = (source) => {
+        Image.getSize(source, this.handleSizeReceived, this.handleSizeFailed);
+    };
+
     handleSizeReceived = (width, height) => {
+        if (!this.mounted) {
+            return;
+        }
+
         this.setState({
             width,
             height
         });
     };
 
     handleSizeFailed = () => {
+        if (!this.mounted) {
+            return;
+        }
+
         this.setState({
-            width: 0,
-            height: 0
+            failed: true
         });
-    }
+    };
 
     handleLayout = (event) => {
+        if (!this.mounted) {
+            return;
+        }
+
         this.setState({
             maxWidth: event.nativeEvent.layout.width
         });
-    }
+    };
+
+    handleLinkPress = () => {
+        const url = normalizeProtocol(this.props.linkDestination);
+
+        Linking.canOpenURL(url).then((supported) => {
+            if (supported) {
+                Linking.openURL(url);
+            }
+        });
+    };
+
+    handleLinkLongPress = async () => {
+        const {formatMessage} = this.context.intl;
+
+        const config = await mattermostManaged.getLocalConfig();
+
+        let action;
+        if (config.copyAndPasteProtection !== 'true') {
+            action = {
+                text: formatMessage({id: 'mobile.markdown.link.copy_url', defaultMessage: 'Copy URL'}),
```

Comment:
this suppose to be an image right? why not set the menu item as `Copy Image URL` ?

Reply:
This is for the same behaviour as above since this image is part of a link